AutoCollection/WebSnippet.ts

```diff
@@ -68,13 +69,98 @@ class WebSnippet {
+        } catch (err) {
+            Logging.info("get web snippet ikey error: " + err);
+        }
+        return iKey;
+    }
+
+    private _getWebInstrumentationReplacedStr() {
+        let configStr = this._getClientWebInstrumentationConfigStr(this._clientWebInstrumentationConfig);
+        let osStr = this._getOsString();
+        let rpStr = this._getResourceProvider();
+        let snippetReplacedStr = `${this._webInstrumentationIkey}\",\r\n${configStr} disableIkeyDeprecationMessage: true,\r\n sdkExtension: \"${rpStr}${osStr}r_n_`;
+        let replacedSnippet = webSnippet.replace("INSTRUMENTATION_KEY", snippetReplacedStr);
+        if (this._clientWebInstrumentationSrc) {
+            return replacedSnippet.replace(`${Constants.WEB_INSTRUMENTATION_DEFAULT_SOURCE}.2.min.js`,this._clientWebInstrumentationSrc);
+        }
+        return replacedSnippet;
+    }
+
+    private _getOsString() {
+        // default is unknown OS
+        let osStr = "u";
+        try {
+            if (os && os.type()) {
+                let osType = os.type();
+                if (osType === OS_WINDOWS) {
+                    osStr = "w"
+                }
+                if (osType === OS_LINUX) {
+                    osStr = "l"
+                }
+            }
+        } catch (err) {
+            Logging.info("get OS type err: " + err)
         }
-        return iKeyCode;
+        return osStr;
+    }
+
+    private _getClientWebInstrumentationConfigStr(config: any) {
+        let configStr = "";
+        try {
+            if (typeof config  === "object") {
+                let keys = Object.keys(config);
+                keys.forEach((key) =>{
+                    let val = config.key || config[key];
+                    let entry = "";
+                    // NOTE: users should convert object/function to string themselves
+                    // Type "function" and "object" will be skipped!
+                    switch(typeof val) {
+                        case "function":
+                            break;
+                        case "object":
+                            break;
+                        case "string":
+                            entry = ` ${key}: \"${val}\",\r\n`;
+                            configStr += entry;
+                            break;
+                        default:
+                            entry = ` ${key}: ${val},\r\n`;
+                            configStr += entry;
+                            break;
+                    }
+                    
+                })
+            }
+
+        } catch (e) {
+            // if has any errors here, web Instrumentation will be disabled.
+            this._isEnabled = false;
+            Logging.info("Parse client web instrumentation error. Web Instrumentation is disabled");
+        }
+        return configStr;
+    }
+
+    private _getResourceProvider() {
+        let resourceProvider = "u";
+        if (process.env.WEBSITE_SITE_NAME) { // Web apps
```

Comment:
WEBSITE_SITE_NAME should exist in both Web Apps and Functions, is not the case anymore?